pkg/crdpuller/discovery.go

```diff
@@ -142,6 +142,9 @@ func (sp *schemaPuller) PullCRDs(context context.Context, resourceNames ...strin
 				continue
 			}
 
+			// pull only the first resource when there is a name conflict (example: pods can be pods or pods.metrics)
+			resourcesToPull.Delete(groupResource.String(), apiResource.Name)
```

Comment:
I don't think this is the correct fix for the issue. In your example, pods.metrics.k8s.io is a resource that is distinct from pods (from core/v1).

Reply:
One alternative approach is to use a RESTMapper to resolve a partially-specified resource name (e.g. "pods") to the match having the highest priority - see https://github.com/kcp-dev/kcp/pull/369/files?w=1 for an example (note, this is not an efficient implementation, but rather the quickest thing I could put together to demonstrate). Hope this helps!

Reply:
Yes it works, thanks. It is better like this. I updated this PR so that the pullAllResources flag is still running. 

If we follow this route, there is an other place where we need to fix this: https://github.com/kcp-dev/kcp/blob/main/pkg/reconciler/cluster/cluster.go#L95

I am not sure I should add a discovery client here and almost the same code. WDYT?

Also, what should be an efficient implementation? Something like using an informer for SchemaGroupVersion ?